features/telemetry/protocols/netflow/parser/src/test/java/org/opennms/netmgt/telemetry/protocols/netflow/parser/session/UdpSessionManagerTest.java

```diff
@@ -159,6 +163,46 @@ private void testSessionKeys(final UdpSessionManager.SessionKey sessionKey1, fin
         Assert.assertEquals(shouldMatch, result.contains(new StringValue("additionalField2", Optional.empty(), "additionalValue2")));
     }
 
+    /**
+     * see NMS-13539
+     */
+    @Test
+    public void optionsRemovalTest() {
+        final UdpSessionManager.SessionKey sessionKey = new Netflow9UdpParser.SessionKey(remoteAddress1.getAddress(), localAddress1);
+
+        final UdpSessionManager udpSessionManager = new UdpSessionManager(Duration.ofMinutes(0), () -> new SequenceNumberTracker(32));
+        final Session session = udpSessionManager.getSession(sessionKey);
+
+        final List<Scope> scopes = new ArrayList<>();
+        scopes.add(scope("scope1", null));
+        scopes.add(scope("scope2", null));
+
+        final List<Field> fields = new ArrayList<>();
+        fields.add(field("field1", null));
+        fields.add(field("field2", null));
+
+        final Template template = Template.builder(100, Template.Type.OPTIONS_TEMPLATE).withFields(fields).withScopes(scopes).build();
+        session.addTemplate(observationId1, template);
+
+        final List<Value<?>> scopesValue = new ArrayList<>();
+        scopesValue.add(value("scope1", "scopeValue1"));
+        scopesValue.add(value("scope2", "scopeValue2"));
+
+        final List<Value<?>> fieldsValue = new ArrayList<>();
+        fieldsValue.add(value("additionalField1", "additionalValue1"));
+        fieldsValue.add(value("additionalField2", "additionalValue2"));
+
+        session.addOptions(observationId1, templateId1, scopesValue, fieldsValue);
+
+        Assert.assertTrue(udpSessionManager.options.keySet().contains(new UdpSessionManager.TemplateKey(sessionKey, observationId1, template.id)));
```

Comment:
```suggestion
        assertThat(udpSessionManager.options.keySet(), hasItem(new UdpSessionManager.TemplateKey(sessionKey, observationId1, template.id)));
```

would yield nice errors on failure than "expected true, but got false"

Reply:
done